cmd/processor/app/processor.go

```diff
@@ -649,3 +657,49 @@ func (p *Processor) setWorkersStatus(triggerInstance trigger.Trigger, status sta
 
 	return nil
 }
+
+// creates a signal handler, so on signal processor gracefully terminated
+func (p *Processor) handleSignals() {
+	var captureSignal = make(chan os.Signal, 1)
+	signal.Notify(captureSignal, syscall.SIGTERM, syscall.SIGABRT, syscall.SIGINT)
+	p.actOnSignal(<-captureSignal)
+}
+
+func (p *Processor) actOnSignal(signal os.Signal) {
+	p.logger.WarnWith("", "Got system signal", signal.String())
+	p.terminateAllTriggers()
+	os.Exit(0)
```

Comment:
I would not exit from a specific function as it i snot a good practice. you may finish your work here, and finish up here and find the way to signal the processor blocking wait to free up so its process can be restarted. see  https://github.com/nuclio/nuclio/blob/14630f3e8fbc40ffc1125e477a350ee9d865e044/cmd/processor/app/processor.go#L228